src/xopen/__init__.py

```diff
@@ -67,6 +68,29 @@ def _available_cpu_count():
         return 1
 
 
+def _can_read_concatenated_gz(program: str) -> bool:
+    """
+    Function to test igzip's bug as reported here: https://github.com/intel/isa-l/issues/143
+    Instead of elaborate version string checking once the problem is fixed, it
+    is much easier to use this, "proof in the pudding" type of evaluation.
+    The used test gzip file was created by:
+    echo -n "AB" | gzip -c > concatenated.gz
+    echo -n "CD" | gzip -c >> concatenated.gz
+    echo -n "EF" | gzip -c >> concatenated.gz
+    echo -n "GH" | gzip -c >> concatenated.gz
+    gzip and pigz can handle this without issue. But bugged version of igzip
+    will only extract the first "AB" block.
+    """
+    test_gz = pathlib.Path(__file__).parent / "resources" / "concatenated.gz"
```

Comment:
I would find it a bit nicer to create this file on the fly in a temporary directory instead of distributing it as a resource. Then you also don’t need to explain how you created the file in a comment since that will be documented right in the code. You also don’t need to concatenate four blocks, just two is enough.

Also, the remainder of the comment should be placed at the location where the function is called since it offers more general functionality than just checking igzip (even the name that you chose implies that already).

Reply:
Done.